src/storage/v2/inmemory/replication/recovery.cpp

```diff
@@ -98,7 +98,7 @@ uint64_t ReplicateCurrentWal(const utils::UUID &main_uuid, const InMemoryStorage
   stream.AppendFilename(wal_file.Path().filename());
   utils::InputFile file;
   MG_ASSERT(file.Open(wal_file.Path()), "Failed to open current WAL file at {}!", wal_file.Path());
-  const auto [buffer, buffer_size] = wal_file.CurrentFileBuffer();
+  const auto [buffer, buffer_size] = wal_file.DurableBuffer();  // Send only the fsynced part of the buffer.
```

Comment:
Why write the buffer at all? stream.AppendFileData(&file); would contain all that is durable